core/domain/exp_jobs_one_off.py

```diff
@@ -324,7 +348,70 @@ def map(item):
                 html_validation_service.validate_math_tags_in_html(html_string))
             if len(error_list) > 0:
                 key = (
-                    'exp_id: %s, exp_status: %s' % (
+                    'exp_id: %s, exp_status: %s failed validation.' % (
+                        item.id, exploration_status))
+                value_dict = {
+                    'state_name': state_name,
+                    'error_list': error_list,
+                    'no_of_invalid_tags': len(error_list)
+                }
+                yield (key, value_dict)
+
+    @staticmethod
+    def reduce(key, values):
+        yield (key, values)
+
+
+class ExplorationMockMathMigrationOneOffJob(jobs.BaseMapReduceOneOffJobManager):
+    """Job that migrates all the math tags in the exploration to the new schema
+    but does not save the migrated exploration. This job is used to verify
+    that the actual migration will be possible for all the explorations.
+    """
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [exp_models.ExplorationModel]
+
+    @staticmethod
+    def map(item):
+        if item.deleted:
+            return
+
+        exploration = exp_fetchers.get_exploration_from_model(item)
+        exploration_status = (
+            rights_manager.get_exploration_rights(
+                item.id).status)
+        for state_name, state in exploration.states.items():
+            try:
+                converted_state_dict = (
+                    state_domain.State.convert_html_fields_in_state(
+                        state.to_dict(),
+                        html_validation_service.
+                        add_math_content_to_math_rte_components))
+                with mock_swap(html_cleaner, 'clean', lambda html: html):
+                    converted_state = (
+                        state_domain.State.from_dict(converted_state_dict))
+                    converted_state.validate(exploration.param_specs, True)
+            except Exception as e:
```

Comment:
Presumably this won't include the stacktrace from "below". You might want to make sure it does -- we've faced debuggability issues in the past due to it not being available.

Reply:
I've removed this part